pkg/ccl/sqlproxyccl/proxy_handler.go

```diff
@@ -451,6 +476,61 @@ func (handler *proxyHandler) handle(ctx context.Context, incomingConn net.Conn)
 	}
 }
 
+// handleCancelRequest handles a pgwire query cancel request by either
+// forwarding it to a SQL node or to another proxy.
+func (handler *proxyHandler) handleCancelRequest(cr *proxyCancelRequest, allowForward bool) error {
+	const timeout = 2 * time.Second
+	if ci, ok := handler.cancelInfoMap.getCancelInfo(cr.SecretKey); ok {
+		if !ci.clientAddr.(*net.TCPAddr).IP.Equal(cr.ClientIP) {
+			// If the IP associated with the cancelInfo does not match the IP from
+			// which the request came, then ignore it.
+			return errors.Errorf("mismatched client IP for cancel request")
+		}
+		var crdbAddr net.Addr
```

Comment:
nit: what do you think of extracting the cancel logic into a helper on cancel info?

e.g.
ci.SendCancel(ctx)

So the implementation of this looks something like:
```
ctx, cacnel := context.WithTimeout(ctx, time.Second * 2)
if ci, ok := handler.cancelInfoMap.getCancelInfo(cr.SecretKey {
   if err := ci.SendCancel(ctx); err != nil {
      // ...
   }
   return nil
}
```
// logic for forward

Reply:
makes a lot of sense, thanks!